### PR TITLE
pkg: Add go-oidc-agent

### PR DESCRIPTION
**Pull request by dave-tucker**
This brings in go-oidc-agent from
https://github.com/redhat-et/go-oidc-agent and its dependent pkgs. It was originally incubated out of tree, but since we're the only user it makes sense for it to co-exist here.

### CODE REVIEW

File: .markdownlint-cli2.yaml [hunk 1]
```diff
@@ -2,3 +2,5 @@ config:
   line-length: false
   no-emphasis-as-header: false
   first-line-heading: false
+ignores:
+  - pkg
```
Comment (russellb) [inline .markdownlint-cli2.yaml:6]: Going to fix this after importing?

File: pkg/oidcagent/pkg/cookie/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# Cookie store for [Session](https://github.com/go-session/session)
```
Comment (russellb) [inline pkg/oidcagent/pkg/cookie/README.md:1]: Looks like you imported this twice? once as `pkg/cookie` and once as `pkg/oidcagent/pkg/cookie` ?

File: pkg/oidcagent/pkg/ginsession/LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (russellb) [inline pkg/oidcagent/pkg/ginsession/LICENSE:1]: This one is imported twice too

File: pkg/oidcagent/pkg/cookie/README.md (hunk 2 above)
Comment (dave-tucker) [reply pkg/oidcagent/pkg/cookie/README.md]: oops. fixed.

File: .markdownlint-cli2.yaml (hunk 1 above)
Comment (dave-tucker) [reply .markdownlint-cli2.yaml]: Maybe. Added more detail to the commit message.
Some of the packages that we have are forks of older, unmaintained packages.
Not sure if we should edit the README's there or not.

**Comment by russellb** [comment]
> @russellb @chirino this now includes hiram's patch to go-oidc-agent, and therefore I did some git magic to preserve the git history. The last 4 commits are purely organizational/ci related, no go-oidc-agent code was changed.

Nice magic!

> I plan on adding another commit as a follow-up to rename a few APIs, then we can rebase #559

Adding another commit to this PR? Or another PR after this one? I wasn’t sure if you meant to hold off on merging.

**Comment by dave-tucker** [comment]
>  Or another PR after this one? 

I meant that 😆

File: .markdownlint-cli2.yaml [hunk 4]
```diff
@@ -2,3 +2,8 @@ config:
   line-length: false
   no-emphasis-as-header: false
   first-line-heading: false
+paths:
+ - "**/md"
```
Comment (russellb) [inline .markdownlint-cli2.yaml:6]: This is different from what was in the command line. Did you verify it still works?

Comment (dave-tucker) [reply .markdownlint-cli2.yaml:6]: well the lint is clear, but this should be `**/*.md`. Will fix it now.

Comment (dave-tucker) [reply .markdownlint-cli2.yaml]: With fix applied:
```
markdownlint-cli2 v0.6.0 (markdownlint v0.27.0)
Finding: **/*.md !pkg !**/node_modules/*
Linting: 21 file(s)
Summary: 0 error(s)
```

**Pull request merged**